test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -243,18 +224,21 @@ void testBacktransformationEigenv(comm::CommunicatorGrid grid, SizeType m, SizeT
   for (SizeType j = nr_reflector - 1; j >= 0; --j) {
     const GlobalElementIndex v_offset{j + mb, j};
     auto tau = tausloc[j];
-    lapack::larf(lapack::Side::Left, m - mb - j, n, v.ptr(v_offset), 1, tau,
-                 c.ptr(GlobalElementIndex{j + mb, 0}), c.ld());
+    if (m != 0 && n != 0) {
```

Comment:
is this check needed?

Reply:
I obtain an error like this, if I remove the check:
```
terminate called without an active exception
[ERROR] /home/lara/Documents/DLAF/DLA-Future_eigenv_bt_distr/test/include/dlaf_test/matrix/matrix_local.h:110 : dlaf::SizeType dlaf::matrix::test::MatrixLocal<const T>::elementLinearIndex(const GlobalElementIndex&) const [with T = float; dlaf::SizeType = long int; dlaf::GlobalElementIndex = dlaf::common::Index2D<long int, dlaf::matrix::GlobalElement_TAG>]
index.isIn(size())
(1, 0)
(3, 0)
```
can I get rid of this in a different way?

Reply:
Ok, I was missing a detail, but IMHO it can be improved (if I'm not missing something again 😄).

The check above is to skip `m = 0 || n == 0`.

Since `nr_reflector` is strictly related to `m`, with `m == 0` it should be 0 too, so the loop won't run at all, and there is no need to check it.

When `n == 0`, it means there is nothing to apply the backtransformation to, so I don't see the reason to compute taus/reflectors.

Reply:
great! it definitely looks better now.
what do you think?